### DISCUSSION TITLE
Produce type directly from std::io::Read

### DISCUSSION THREAD

**Discussion by rbran** [sharksforarms/deku#318] [Ideas]
First of all, thanks for the crate, it's amazing.

While using it, I found that i keep writing something like this:
```rust
let mut value = [0u8; 4];
reader.read_exact(&mut value)?;
let (_, value) = DekuValue::from_bytes((&value, 0))?;
```
And I wish that I could do something like this instead:
```rust
let (_, value) = DekuValue::from_reader(reader)?;
```
Proposal: https://gist.github.com/rbran/e7814fd6694ee37d54243236b77ff3cf

**Comment by wcampbell0x2a** [comment]
There is an issue to add this, just haven't implemented it: https://github.com/sharksforarms/deku/issues/105
